src/function.rs

```diff
@@ -278,37 +301,122 @@ where
   }
 }
 
+pub struct FunctionBuilder<'s> {
+  callback: FunctionCallback,
+  data: Option<Local<'s, Value>>,
+  length: i32,
+  constructor_behavior: ConstructorBehavior,
+  side_effect_type: SideEffectType,
+  name: Option<Local<'s, String>>,
+}
```

Comment:
I think one of rusty_v8's goals is to not provide too much syntactic sugar on top of the V8 API (i.e., map 1-to-1 as closely as possibly) and this builder pattern is kind of at odds with that.

More idiomatic (if not perhaps ergonomic) would be something like this:
```rust
impl Function {
  pub fn new( // or e.g. new_with_options
    scope: &mut HandleScope<'s>,
    callback: impl MapFnTo<FunctionCallback>,
    data: Option<Local<Value>>,
    length: Option<i32>,
    behavior: Option<ConstructorBehavior>,
    side_effect_type: Option<SideEffectType>,
  ) -> Option<Local<'s, Function>> {
    // ...
    let length = length.unwrap_or(0);
    let constructor_behavior = constructor_behavior.unwrap_or(ConstructorBehavior::Allow);
    let side_effect_type = side_effect_type.unwrap_or(SideEffectType::HasSideEffect);
    // ...
  }

  pub fn set_name(&mut self, name: Local<String>) {
    // ...
  }
}
```

Reply:
yeah idk... having them all in one signature like that is pretty horrifying. I guess the question is whether this library is more of a v8_sys or v8.